### PR TITLE
legacy: Create app CR for per cluster app-operator instance

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/14163

Same as https://github.com/giantswarm/cluster-operator/pull/1336 but for legacy.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/resource/appfinalizer/delete.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+				return microerror.Mask(err)
+			}
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("removed finalizer for app %#q", app.Name))
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finalizer already removed for app %#q", app.Name))
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking all apps are deleted in namespace %#q", clusterConfig.ID))
```
Comment (rossf7) [inline service/controller/resource/appfinalizer/delete.go:66]: For legacy we check all the apps are deleted and if not keep the finalizers. 

Without this the cluster namespace stayed in terminating but the azureclusterconfig CR had been deleted. I didn't see that problem with the master branch as it keeps the finalizers.

File: service/controller/resource/appversionlabel/create.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	var apps []*v1alpha1.App
 	{
-		r.logger.LogCtx(ctx, fmt.Sprintf("finding optional apps for tenant cluster %#q", clusterConfig.ID))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding optional apps for tenant cluster %#q", clusterConfig.ID))
```
Comment (rossf7) [inline service/controller/resource/appversionlabel/create.go:25]: Just fixing the logging here.

**Comment by rossf7** [comment]
Tested on gremlin.

File: service/controller/resource/appfinalizer/delete.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	for _, app := range list.Items {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("removing finalizer for app %#q", app.Name))
+
+		index := getFinalizerIndex(app.Finalizers)
+		if index >= 0 {
+			patches := []patch{
+				{
+					Op:   "remove",
+					Path: fmt.Sprintf("/metadata/finalizers/%d", index),
```
Comment (tomahawk28) [inline service/controller/resource/appfinalizer/delete.go:47]: Can we simply remove all the finalizers?

Comment (rossf7) [reply service/controller/resource/appfinalizer/delete.go:47]: I'd prefer to keep it the same as the master branch.

Also app-operator has the migration logic to set the finalizer to the one being removed here. So I think it's safer to just remove that finalizer.

**Review by tomahawk28 — APPROVED** [review]
LGTM

**Pull request merged**